app/os_android.go

```diff
@@ -951,3 +1012,40 @@ func Java_org_gioui_Gio_scheduleMainFuncs(env *C.JNIEnv, cls C.jclass) {
 }
 
 func (_ ViewEvent) ImplementsEvent() {}
+
+// StartForeground starts a foreground service and returns a channel that stops the service when closed or error.
+func StartForeground(title, text string) (chan struct{}, error) {
+	var closeChan chan struct{}
+	errChan := make(chan error)
+
+	// get a handle on the startService, stopService methods of GioForegroundService
+	// run everything in a goroutine so that start/stop calls are from the same thread
+	go func() {
+		runInJVM(javaVM(), func(env *C.JNIEnv) {
+			startForegroundService := getStaticMethodID(env, android.gioCls,
+				"startForegroundService", "(Landroid/content/Context;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;")
+
+			intent, err := callStaticObjectMethod(env, android.gioCls,
+				startForegroundService,
+				jvalue(android.appCtx),
+				jvalue(javaString(env, "org/gioui/GioForegroundService")),
```

Comment:
You provide the name of the foreground service class, but that's a constant. I had expected "org/gioui/GioActivity", optionally fetched from meta-data.

Reply:
hm yeah. not sure what happened here.